catalog/app/utils/LogicalKeyResolver.tsx

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react'
+
+interface S3Handle {
```

Comment:
> Because it can be confusing with handleS3, handleClick

i disagree, bc in case of `handleThing` "handle" is a verb and in case of `thingHandle` "handle" is a noun indicating that one can somehow reference that "thing" with that "handle", so i think it's perfectly valid and absolutely non-confusing.

better name would be smth like "S3ObjectRef", and i plan rewriting and unifying all the s3-related logic into coherent typed modules somewhere soon-ish.

Reply:
Later in code, you are using `HandleResolver` it can be read as "handle-verb resolver" and "handle-noun resolver"